src/main/features/claim/routes/pay-by-account.ts

```diff
@@ -122,9 +124,9 @@ export default express.Router()
         } else {
           const payClient: PayClient = await getPayClient()
           const paymentResponse: PaymentResponse = await payClient.create(
-            res.locals.user,
+            user,
             draft.document.feeAccount.reference,
-            draft.document.externalId,
+            await ClaimStoreClient.retrievePaymentReference(draft.document.externalId, user),
```

Comment:
That seems alright under condition that `ClaimStoreClient.retrievePaymentReference` is idempotent and can be called many times. If such condition is violated then users would not be able to recover when draft updated failed.

Reply:
it is idempotent given the first call finished, I wouldn't recommend calling it 3 times at the same time though